drv/sidecar-mainboard-controller/src/tofino2.rs

```diff
@@ -231,3 +232,435 @@ impl Sequencer {
         })
     }
 }
+
+bitfield! {
+    #[derive(Copy, Clone, PartialEq, FromPrimitive, AsBytes)]
+    #[repr(C)]
+    pub struct DebugPortState(u8);
+    pub send_buffer_empty, set_send_buffer_empty: 0;
+    pub send_buffer_full, _: 1;
+    pub receive_buffer_empty, set_receive_buffer_empty: 2;
+    pub receive_buffer_full, _: 3;
+    pub request_in_progress, set_request_in_progress: 4;
+    pub address_nack_error, set_address_nack_error: 5;
+    pub byte_nack_error, set_byte_nack_error: 6;
+}
+
+#[derive(Copy, Clone, PartialEq, FromPrimitive, AsBytes)]
+#[repr(u8)]
+pub enum DebugRequestOpcode {
+    LocalWrite = 0,
+    LocalRead = 32,
+    DirectWrite = 128,
+    DirectRead = 160,
+    IndirectWrite = 192,
+    IndirectRead = 224,
+}
+
+impl From<DebugRequestOpcode> for u8 {
+    fn from(opcode: DebugRequestOpcode) -> Self {
+        opcode as u8
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, FromPrimitive, AsBytes)]
+#[repr(u8)]
+pub enum DirectBarSegment {
+    Bar0 = 0,
+    Msi = 1,
```

Comment:
It's weird to me that we explicitly specify values, but those values _aren't_ what we use for `impl From<DirectBarSegment> for u32`.

I'd vote to either
(a) make this a simple `enum` (no `repr` or assigned values), _or_
(b) make this `#[repr(u32)]` and use the `u32` values below

(leaning slightly towards the latter)

Reply:
Yes, this is only used in the context where you want that base address. So using the desired address as the representation makes sense.